### PR TITLE
[KYUUBI #3107] [Subtask] DorisSQLEngine - Add process builder

### PR DESCRIPTION
**Pull request by zhaomin1423**
### _Why are the changes needed?_

close https://github.com/apache/incubator-kyuubi/issues/3107#issue-1311302964
### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by zhaomin1423** [comment]
image

**Comment by pan3793** [comment]
So far LGTM, would u please add an it module for jdbc engine?

**Comment by zhaomin1423** [comment]
> So far LGTM, would u please add an it module for jdbc engine?

Yes, I plan to do it in a new pr.

**Comment by zhaomin1423** [comment]
image

Redacted the kyuubi.engine.jdbc.connection.password using *********(redacted) in the log. @pan3793

**Review by yaooqinn — APPROVED** [review]

**Comment by zhaomin1423** [comment]
Because added all jars to external/jdbc, https://github.com/apache/incubator-kyuubi/pull/3146, so the class paths are modify.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/jdbc/JdbcProcessBuilder.scala [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      conf.get(ENGINE_JDBC_CONNECTION_URL).nonEmpty,
+      s"Jdbc server url can not be null! Please set ${ENGINE_JDBC_CONNECTION_URL.key}")
+    val buffer = new ArrayBuffer[String]()
+    buffer += executable
+
+    val memory = conf.get(ENGINE_JDBC_MEMORY)
+    buffer += s"-Xmx$memory"
+
+    val javaOptions = conf.get(ENGINE_JDBC_JAVA_OPTIONS)
+    if (javaOptions.isDefined) {
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/jdbc/JdbcProcessBuilder.scala:73]: nit: (javaOptions.foreach ...)

**Review by pan3793 — APPROVED** [review]

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/jdbc/JdbcProcessBuilder.scala (hunk 1 above)
Comment (zhaomin1423) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/jdbc/JdbcProcessBuilder.scala]: Thanks, updated.

**Comment by yaooqinn** [comment]
please use dev/merge_kyuubi_pr.sh to merge, and update assignee and milestone too

**Comment by zhaomin1423** [comment]
> please use dev/merge_kyuubi_pr.sh to merge, and update assignee and milestone too
> 

Thanks for your reminder.

**Pull request merged**